### PR TITLE
Added self documenting application to preoject

### PR DESCRIPTION
**Pull request by cutandpastey**
I wanted to dig a little deeper into Angular and learn some of the internals over the weekend. I was also running into situations in which having all the components displayed within the index page proved hard to debug so I've turned `index.html` into an SPA.

screen shot 2015-03-23 at 10 21 13

screen shot 2015-03-23 at 10 21 23

This should mean that the project becomes self documenting. Every component should have a page and code should be displayed within:

``` html
<app-demo>
  <app-demo-code>
    <!-- Your code goes here -->
  </app-demo-code>
</app-demo>
```

which produces a code block within the page (see images above).

### CODE REVIEW

**Comment by robinedman** [comment]
Nice stuff!

My one suggestion to simplify a bit would be this: The controllers don't really seem to be used. I'm wondering whether we might as well just use a controller called `AppCtrl` or similar. None of the components rely on things on the scope so I don't think it should be a problem.

**Comment by cutandpastey** [comment]
Yeah, I just copied the example from the angular docs. Not sure if you can omit the controller from the router.

File: index.html [hunk 1]
```diff
@@ -3,106 +3,58 @@
...
 
+    <gu-column span="8" gu-box="secondary" skin="large">
+      Main section
+    </gu-column>
 
+  </gu-row>
+  <gu-row class="content">
+    <gu-column class="sidebar" span="2" gu-box="tertiary">
+      <ul>
+        <li><a href='#/components/icons'>Icons</a></li>
```
Comment (jamespamplin) [inline index.html]: Quotes inconsistent ;)

**Comment by jamespamplin** [comment]
Cool stuff :+1:

**Comment by robinedman** [comment]
You can instantiate a controller several times so just passing the same controller in into every route should work fine.

**Comment by theefer** [comment]
Have you tried without any controller? Seems tedious to have lots of no-op boilerplate.

**Comment by cutandpastey** [comment]
Yeah, it works. TBQH can't believe I didn't just try this anyway. Will fix in another PR.

**Pull request merged**